data/components/connection-filter.js

```diff
@@ -0,0 +1,67 @@
+/* See license.txt for terms of usage */
+
+define(function(require, exports/*, module*/) {
+
+"use strict";
+
+// Dependencies
+const React = require("react");
+
+// WebSockets Monitor
+const { filterFrames } = require("../actions/frames");
+
+// Shortcuts
+const { span, select, option } = React.DOM;
+
+/**
+ * This component renders a select element when unique
+ * webSocket connections > 1. Using this dropdown, the
+ * user can select the connection ID they are interested
+ * in seeing, and a reducer should filter out all other
+ * connections.
+ */
+var ConnectionFilter = React.createClass({
+/** @lends ConnectionFilter */
+
+  displayName: "ConnectionFilter",
+
+  handleChange: function(e) {
+    const { value } = e.target;
+    const currentFilter = this.props.frames.filter;
+
+    // Dispatch new filter, merging with old filter to
+    // retain text filter alongside this filter.
+    this.props.dispatch(filterFrames(
+      Object.assign({}, currentFilter, {
+        webSocketSerialID: value ? Number(value) : null
```

Comment:
Just a nit: you may test `value !== null`, so you handle when the ID is 0 (if that's possible, I agree that would be extremely rare, but let's keep the devil away from this code :), and we don't know if the platform won't change the way the IDs are generated)